lib/dtutils/file.lua

```diff
@@ -0,0 +1,523 @@
+--[[
+
+    dtutils/file.lua - common darktable lua file functions
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+    Copyright (C) 2016 Tobias Jakobs
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local dtutils_file = {}
+dtutils_file.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "License"},
+  Name = [[dtutils.file - common darktable lua file functions]],
+  Synopsis = [[local df = require "lib/dtutils.file"]],
+  Description = [[dtutils.file provides common file manipulation functions used in
+  constructing Darktable lua scripts]],
+  License = [[This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.]],
+  functions = {}
+}
+
+local dt = require "darktable"
+
+local log = require "lib/libLog"
+
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("dtutils.file",dt.configuration.config_dir.."/lua/locale/")
```

Comment:
How about we break this up into 2 problems?  

Problem 1 is where do we put the .mo files.  I think we can solve that one now.  My vote is a locale directory in the lua-scripts directory.  That way we don't have all the different language directories scattered through out the base directory.

Problem 2 is how to get the .mo files to where we decide they should belong.  I'm still thinking on that one, but I have an idea...  ;)